### PR TITLE
New `LinearOperator`: `CysparseLinearOperator

### PR DESCRIPTION
**Pull request by syarra**
Added a new operator.

### CODE REVIEW

File: pykrylov/linop/linop.py [hunk 1]
```diff
@@ -720,6 +720,45 @@ def matvec_transp(y):
...
+
+    nargout, nargin = A.shape
+    symmetric = A.is_symmetric
+
+    def matvec(x):
+        if x.shape != (nargin,):
+            msg = 'Input has shape ' + str(x.shape)
+            msg += ' instead of (%d,)' % nargin
+            raise ShapeError(msg)
+        return A.matvec(x)
```
Comment (dpo) [inline pykrylov/linop/linop.py]: Does `CySparse` already perform an argument check? If so, there's no need to perform it twice.

Also, can't we write `A * x`?

File: pykrylov/linop/linop.py [hunk 2]
```diff
@@ -720,6 +720,45 @@ def matvec_transp(y):
...
+
+    def matvec_transp(y):
+        if y.shape != (nargout,):
+            msg = 'Input has shape ' + str(y.shape)
+            msg += ' instead of (%d,)' % nargout
+            raise ShapeError(msg)
+        return A.matvec_transp(y)
```
Comment (dpo) [inline pykrylov/linop/linop.py]: `A.T * y`?

File: pykrylov/linop/linop.py [hunk 3]
```diff
@@ -720,6 +720,45 @@ def matvec_transp(y):
...
+
+    def matvec_adj(y):
+        if y.shape != (nargout,):
+            msg = 'Input has shape ' + str(y.shape)
+            msg += ' instead of (%d,)' % nargout
+            raise ShapeError(msg)
+        return A.matvec_htransp(y)
```
Comment (dpo) [inline pykrylov/linop/linop.py]: `A.H * y`?

**Comment by dpo** [comment]
What are those 14 commits?

**Comment by syarra** [comment]
Wrong rebase, so I deleted this branch.

**Pull request closed without merging**